akka-cluster/src/main/scala/akka/cluster/Cluster.scala

```diff
@@ -317,8 +319,10 @@ class Cluster(val system: ExtendedActorSystem) extends Extension {
    * When it has successfully joined it must be restarted to be able to join another
    * cluster or to join the same cluster again.
    */
-  def joinSeedNodes(seedNodes: immutable.Seq[Address]): Unit =
+  def joinSeedNodes(seedNodes: immutable.Seq[Address]): Unit = {
+    seedNodes.foreach(checkHostCharacters)
```

Comment:
should also be added to `akka.cluster.typed.JoinSeedNodes` message (constructor check)